jOOQ/src/main/java/org/jooq/impl/InsertQueryImpl.java

```diff
@@ -278,10 +279,21 @@ else if (onDuplicateKeyIgnore) {
                 case POSTGRES_9_5:
                 case POSTGRES: {
                     toSQLInsert(ctx);
-                    ctx.formatSeparator()
-                       .start(INSERT_ON_DUPLICATE_KEY_UPDATE)
-                       .keyword("on conflict do nothing")
-                       .end(INSERT_ON_DUPLICATE_KEY_UPDATE);
+                    ctx.formatSeparator().start(INSERT_ON_DUPLICATE_KEY_UPDATE).keyword("on conflict");
+
+                    if (onConflict != null && onConflict.size() > 0) {
+                        if (onConflict.size() != 1) {
+                            LOGGER.info(
+                                    "ON DUPLICATE KEY IGNORE accept only one field for unique/exclusion constraint, " +
+                                            "let's take the first one on the list {}", onConflict.get(0).getName());
+                        }
+                        ctx.sql(" (");
+                        boolean qualify = ctx.qualify();
+                        ctx.qualify(false).visit(onConflict.get(0)).qualify(qualify);
+                        ctx.sql(") ");
+                    }
+
+                    ctx.sql(" do nothing").end(INSERT_ON_DUPLICATE_KEY_UPDATE);
```

Comment:
OK, that's been done in a separate commit: https://github.com/jOOQ/jOOQ/pull/5702/commits/b75cf702a8ee272fc7e62adf25c0cf463fb0e131